src/heuristic/HeuristicMapper.cpp

```diff
@@ -16,6 +16,27 @@ void HeuristicMapper::map(const Configuration& configuration) {
               << " not suitable for heuristic mapper!" << std::endl;
     return;
   }
+  if (config.considerFidelity &&
+      architecture.getSingleQubitFidelities().size() == 0) {
+    std::cerr << "No calibration data available for this architecture! "
+              << "Performing mapping without considering fidelity."
+              << std::endl;
+    config.considerFidelity = false;
+  }
+  if (config.considerFidelity && config.lookahead) {
+    std::cerr << "Lookahead is not yet supported for heuristic mapper using "
+                 "fidelity-aware mapping!"
```

Comment:
This should probably state that mapping is still performed, but without lookahead similar to the above warning.